google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/SimpleParser.java

```diff
@@ -0,0 +1,302 @@
+    // TODO: Implement support for quoted identifiers.
+    // TODO: Implement support for identifiers with multiple parts (e.g. my_schema.my_table).
+    if (!hasMoreTokens()) {
+      return Result.NOT_FOUND;
+    }
+    if (!isValidIdentifierFirstChar(sql.charAt(pos))) {
+      return Result.NOT_FOUND;
+    }
+    int startPos = pos;
+    while (pos < sql.length() && isValidIdentifierChar(sql.charAt(pos))) {
+      pos++;
+    }
+    return Result.found(sql.substring(startPos, pos));
+  }
+
+  /**
+   * Eats a single-quoted string. This implementation currently does not support escape sequences.
+   */
+  Result eatSingleQuotedString() {
+    if (!eatToken('\'')) {
+      return Result.NOT_FOUND;
+    }
+    int startPos = pos;
+    while (pos < sql.length() && sql.charAt(pos) != '\'') {
+      if (sql.charAt(pos) == '\n') {
+        return Result.NOT_FOUND;
+      }
+      pos++;
+    }
+    if (pos == sql.length()) {
+      return Result.NOT_FOUND;
+    }
+    return Result.found(sql.substring(startPos, pos++));
+  }
+
+  boolean peekTokens(char... tokens) {
+    return internalEatTokens(/* updatePos = */ false, tokens);
+  }
+
+  /**
+   * Returns true if the next tokens in the SQL string are equal to the given tokens, and advances
+   * the position of the parser to after the tokens. The position is not changed if the next tokens
+   * are not equal to the list of tokens.
+   */
+  boolean eatTokens(char... tokens) {
+    return internalEatTokens(/* updatePos = */ true, tokens);
+  }
+
+  /**
+   * Returns true if the next tokens in the SQL string are equal to the given tokens, and advances
+   * the position of the parser to after the tokens if updatePos is true. The position is not
+   * changed if the next tokens are not equal to the list of tokens, or if updatePos is false.
+   */
+  private boolean internalEatTokens(boolean updatePos, char... tokens) {
+    int currentPos = pos;
+    for (char token : tokens) {
+      if (!eatToken(token)) {
+        pos = currentPos;
+        return false;
+      }
+    }
+    if (!updatePos) {
+      pos = currentPos;
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if the next token is equal to the given character, but does not advance the
+   * position of the parser.
+   */
+  boolean peekToken(char token) {
+    int currentPos = pos;
+    boolean res = eatToken(token);
+    pos = currentPos;
+    return res;
+  }
+
+  /**
+   * Returns true and advances the position of the parser if the next token is equal to the given
+   * character.
+   */
+  boolean eatToken(char token) {
+    skipWhitespaces();
+    if (pos < sql.length() && sql.charAt(pos) == token) {
+      pos++;
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Returns true if the given character is valid as the first character of an identifier. That
+   * means that it can be used as the first character of an unquoted identifier.
+   */
+  private boolean isValidIdentifierFirstChar(char c) {
```

Comment:
The methods `isValidIdentifierFirstChar` and `isValidIdentifierChar` already exist in `AbstractStatementParser`. Is there a way to re-use?